### ISSUE TITLE
Classes decorated with TypeConverterAttribute no longer deserialize in 4.2.4 or above

### ISSUE THREAD

**Issue by cyotek** [aaubry/YamlDotNet#314]
Hello,

I have a project which is currently using the YamlDotNet.Signed package version 4.2.3-pre0454.

One of the objects being deserialized is a `Process` class with assorted properties. However, it is decorated with `[TypeConverter(typeof(ProcessConverter))]` so that a simple string can be used to create an instance the class. Using 4.2.3-pre0454, I can deserialize a YAML document containing process strings absolutely fine.

I upgraded the project to use the latest version of YamlDotNet, but this causes YAML deserialization to fail trying to convert a process string to a concrete object. Sticking breakpoints in my `ProcessConverter` classes shows it is no longer being called.

Downgrading to the release version of 4.2.3 works fine. Upgrading to 4.2.4 breaks again.

I checked the methods of `DeserializerBuilder` but the type conversion methods on this class require `IYamlTypeConvertor` instances rather than my existing `TypeConverter` classes. Doing a search of this repository for `TypeConverterAttribute` brought up the `YamlDotNet.Serialization.Utilities.TypeConverter.RegisterTypeConverter` method but trying to use this to register the type had no impact.

How can I get automatic type conversion working again?

Thanks;
Richard Moss

**Comment by aaubry** [comment]
This looks like a bug. I'm surprised that no unit caught it. I will look into it later today.

**Comment by aaubry** [comment]
Which platform are you using? .NET 4.5 ? dotnet core ?

**Comment by aaubry** [comment]
If you are on dotnet core, please try this version: https://www.nuget.org/packages/YamlDotNet.Signed/4.3.2-issue-3140481
Then let me know if this fixes your problem. Otherwise, can you provide an example of code that reproduces this issue?

**Comment by cyotek** [comment]
Hello,

Apologies for missing that info out; I'm still using .net 4.5. I'll test the new build if it works with it and create a repro sample if not.

Regards
Richard Moss

**Comment by cyotek** [comment]
Hello,

Sorry for the delay in getting back to you. I created a quick sample project (.NET 4.7) tonight using 4.2.3 which worked as expected. Next I tested 4.3.2-issue-3140481 which also worked. Finally I dropped it back to 4.3.1 which promptly crashed. 

I've attached the sample project to this issue, although based on these tests whatever the problem is you've fixed in that special build.

Regards;
Richard Moss

[YamlDotTypeConverter.zip](https://github.com/aaubry/YamlDotNet/files/1895839/YamlDotTypeConverter.zip)
